adoc/chapters/programming_interface.adoc

```diff
@@ -17640,40 +17635,500 @@ member function template.
 
 
 [[swizzled-vec-class]]
-==== Swizzled [code]#vec# class
-
-The [code]#+__swizzled_vec__+# class must define an unspecified temporary which
-provides the entire interface of the SYCL [code]#vec# class template, including
-swizzled member functions, with the additions and alterations described below.
-The member functions of the [code]#+__swizzled_vec__+# class behave as though
-they operate on a [code]#vec# that is the result of the swizzle operation.
-
-  * The [code]#+__swizzled_vec__+# class template must be readable as an r-value
-    reference on the RHS of an expression.
-    In this case the swizzle operation is performed on the RHS of the expression
-    and then the result is applied to the LHS of the expression.
-  * The [code]#+__swizzled_vec__+# class template must be assignable as an
-    l-value reference on the LHS of an expression.
-    In this case the RHS of the expression is applied to the original SYCL
-    [code]#vec# which the [code]#+__swizzled_vec__+# represents via the swizzle
-    operation.
-    Note that a [code]#+__swizzled_vec__+# that is used in an l-value expression
-    may not contain any repeated element indexes.
-+
-For example: [code]#f4.xxxx() = fx.wzyx()# would not be valid.
-  * The [code]#+__swizzled_vec__+# class template must be convertible to an
-    instance of SYCL [code]#vec# with the type [code]#DataT# and number of
-    elements specified by the swizzle member function, if [code]#NumElements >
-    1#, and must be convertible to an instance of type [code]#DataT#, if
-    [code]#NumElements == 1#.
-  * The [code]#+__swizzled_vec__+# class template must be non-copyable,
-    non-moveable, non-user constructible and may not be bound to a l-value or
-    escape the expression it was constructed in.
-    For example [code]#auto x = f4.x()# would not be valid.
-  * The [code]#+__swizzled_vec__+# class template should return
-    [code]#+__swizzled_vec__&+# for each operator inherited from the [code]#vec#
-    class template interface which would return [code]#vec<DataT,
-    NumElements>&#.
+==== The [code]#+__swizzled_vec__+# class
+
+The [code]#+__swizzled_vec__+# class is a view over a [code]#vec# object which
+reflects a swizzle operation.
```

Comment:
I addressed this and also some of your other comments on the introduction in 0271b4e1c1dad3348f526d441ff34ea1a5877aa3.

Here, I used the wording "captures the effects of a swizzle operation without actually performing that operation" instead of "lazy".  This wording also helps address your comment about clarifying the behavior when the `__swizzled_vec__` is neither read nor written.

Reply:
The new wording sounds good, thanks.